tls/s2n_early_data.c

```diff
@@ -130,6 +132,34 @@ S2N_RESULT s2n_early_data_accept_or_reject(struct s2n_connection *conn)
     return S2N_RESULT_OK;
 }
 
+int s2n_config_set_server_max_early_data_size(struct s2n_config *config, uint32_t max_early_data_size)
```

Comment:
Should both config_set_server_max_early_data_size and connection_set_server_max_early_data_size error if mode is not server?

Reply:
For the config: definitely not. The config doesn't know whether it's a client or server config, and it could be both.

For the connection: I considered it, but I'm not sold. We don't do that for other APIs. There literally isn't a S2N_ERR_SERVER_MODE (although I'm adding it in https://github.com/aws/s2n-tls/pull/2682). It doesn't break anything if a client sets the value either, it just won't do anything.

Reply:
Alright that works for me.